src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -90,7 +90,9 @@ def load_arguments(self, _):
                    help='Skip creating the default assignment, which allows the service principal to access resources under the current subscription. '
                         'When specified, --scopes will be ignored. You may use `az role assignment create` to create '
                         'role assignments for this service principal later.')
-        c.argument('show_auth_for_sdk', options_list='--sdk-auth', help='output result in compatible with Azure SDK auth file', arg_type=get_three_state_flag())
+        c.argument('show_auth_for_sdk', options_list='--sdk-auth', arg_type=get_three_state_flag(),
+                   help='Generate a JSON dictionary compatible with Azure SDK. For how to use it with Azure SDK, '
```

Comment:
Still misleading to me, since it's not compatible with SDK as a whole. I would be more explicit.

"SDK has deprecated usage of authentication file. More details here: aka"